core/domain/question_domain.py

```diff
@@ -203,9 +203,39 @@ def _convert_state_v28_dict_to_v29_dict(cls, question_state_dict):
         question_state_dict['solicit_answer_details'] = False
         return question_state_dict
 
+    @classmethod
+    def _convert_state_v29_dict_to_v30_dict(
+            cls, question_state_dict, linked_skill_ids):
+        """Converts from version 29 to 30. Version 30 replaces
+        tagged_misconception_id with tagged_skill_misconception_id, which
+        contains the skill id and misconception id of the tagged misconception,
+        connected by '-'.
+
+        Args:
+            question_state_dict: dict. A dict where each key-value pair
+                represents respectively, a state name and a dict used to
+                initalize a State domain object.
+            linked_skill_ids: list(str). Skill ids linked to the question.
+
+        Returns:
+            dict. The converted question_state_dict.
+        """
+        answer_groups = question_state_dict['interaction']['answer_groups']
+        for answer_group in answer_groups:
+            if answer_group['tagged_misconception_id'] != None:
+                answer_group['tagged_skill_misconception_id'] = (
+                    linked_skill_ids[0] + '-' +
+                    str(answer_group['tagged_misconception_id']))
+            else:
+                answer_group['tagged_skill_misconception_id'] = None
+
+            del answer_group['tagged_misconception_id']
+        return question_state_dict
+
     @classmethod
     def update_state_from_model(
-            cls, versioned_question_state, current_state_schema_version):
+            cls, versioned_question_state, current_state_schema_version,
+            question_model=None):
```

Comment:
I don't know if its a good idea doing it this way. Let's instead always pass the question_model (and pass None in places where we don't need it) instead of setting default params.

Reply:
Actually that's what I was doing before, but @aks681 suggested to change it to this way (if I didn't misunderstand). I'm not sure which way is better.

Reply:
Done.